### PR TITLE
Add tests to improve coverage for tenacity_stop, microagent and json utils

### PR DESCRIPTION
**Pull request by ImBIOS**
This PR adds new tests to improve coverage for several core components:

- Add tests for tenacity_stop.py (100% coverage)
- Add tests for microagent.py (100% coverage)
- Add tests for json.py (93% coverage)

The new tests cover core functionality including:
- Event serialization
- JSON parsing and repair
- Microagent metadata handling
- Tenacity stop conditions

All tests pass and no existing functionality is modified.

### CODE REVIEW

**Comment by ImBIOS** [comment]
it might contain a lot of commits, I recommend using "Squash & Commit" to squash all commits into a single commit.

File: openhands/core/utils/json.py [hunk 1]
```diff
@@ -46,8 +46,13 @@ def loads(json_str, **kwargs):
...
                 response = json_str[start : i + 1]
                 try:
-                    json_str = repair_json(response)
-                    return json.loads(json_str, **kwargs)
+                    # First try to parse as is
+                    try:
+                        return json.loads(response, **kwargs)
+                    except json.JSONDecodeError:
+                        # If that fails, try to repair it
+                        json_str = repair_json(response)
```
Comment (enyst) [inline openhands/core/utils/json.py:54]: This is a little interesting bit... I thought `repair_json` was doing this itself. Did the tests fail without this?

**Comment by ImBIOS** [comment]
> @ImBIOS just want to check in to see if you have this on your radar still?

I'm sorry, but unfortunately, I'll be busy with something for several more weeks. I would be thankful if someone picked up my work and continued it until it was merged.

**Comment by mamoodi** [comment]
After speaking with the team, going to close this for now. Please feel free to reopen a new PR if you have time to complete it.

**Pull request closed without merging**